modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/publisher/APIApplicationPublisherImpl.java

```diff
@@ -113,7 +163,12 @@ private void _unregisterServiceRegistration(
 
 	private static BundleContext _bundleContext;
 
-	private final Map<String, ServiceRegistration<Application>>
-		_headlessBuilderApplicationServiceRegistrationMap = new HashMap<>();
+	private final Map
+		<String,
+		 ObjectValuePair
+			 <ServiceRegistration<Application>,
+			  ServiceRegistration<HeadlessBuilderResourceImpl>>>
+				_headlessBuilderApplicationServiceRegistrationMap =
```

Comment:
Maybe this name should be modified as not it is not only headless builder application service what it contains

Reply:
Done